sdk-base/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationAccuracyRadiusConsumer.kt

```diff
@@ -0,0 +1,26 @@
+package com.mapbox.maps.plugin.locationcomponent
+
+import android.animation.ValueAnimator
+
+/**
+ * Defines the interface for LocationAccuracyRadiusConsumer.
+ */
+interface LocationAccuracyRadiusConsumer {
```

Comment:
could we introduce a `LocationConsumer2` which is declared as 
```
interface LocationConsumer2 : LocationConsumer {
  fun onAccuracyRadiusUpdated(
    vararg radius: Double,
    options: (ValueAnimator.() -> Unit)? = null
  )
  fun onPuckAccuracyRadiusAnimatorDefaultOptionsUpdated(options: ValueAnimator.() -> Unit)
}
```

We have seen similar pattern in the Android lifecycle callback https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/content/ComponentCallbacks2.java, docs [here](https://developer.android.com/reference/android/content/ComponentCallbacks2) 

while the LocationProvider still accepts the LocationConsumer, we could check if it's the extended version in the implementation and call the extended method if it exist? And `LocationConsumer2` is public and could be suggested to end-users if they want to use the extended version with more functions. wdyt? cc @mapbox/maps-android

Reply:
This also gives user a signal that we are iterating on these interfaces, and in the next major release, we could deprecate/remove the first version and directly use the second version instead.